src/adapter/src/catalog/storage.rs

```diff
@@ -385,6 +385,21 @@ async fn migrate(
             })?;
             Ok(())
         },
+        // Role memberships were added to role definitions.
+        //
+        // Introduced in v0.46.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.47.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            txn.roles.update(|_role_key, role_value| {
+                let mut role_value = role_value.clone();
+                if role_value.role.membership.is_none() {
+                    role_value.role.membership = Some(RoleMembership::new());
```

Comment:
Does it need to be a member of its own role?

Reply:
No, the way I've implemented it is that the `RoleMembership` doesn't include itself. This makes it slightly easier to detect circular memberships.